.github/workflows/ci.yml

```diff
@@ -0,0 +1,24 @@
+name: ci
+
+on: [push, pull_request]
+
+env:
+  X_PYTHON_VERSION: "3.12"
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ env.X_PYTHON_VERSION }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ env.X_PYTHON_VERSION }}
+    - name: Install Python dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r docs/docs-requirements.txt
+    - name: Check documentation for errors
+      run: |
+        SPHINXOPTS="-a -E -n -W --keep-going" make -C docs html
```

Comment:
Can you add an empty line at the end of the file please

Reply:
I think this file also has conflicts with the current state of the main branch

Reply:
Resolving the conflicts was a bit painful as I basically overwrote the ci.yml file in my branch. I rebased onto main and force pushed whilst removing some commits. I rechecked and the results look good. It can also be merged without conflicts now.

Reply:
Thank you for your hassle!